nexus/db-queries/src/db/datastore/sled.rs

```diff
@@ -82,7 +86,10 @@ impl DataStore {
                         &sled_update.id().to_string(),
                     ),
                 )
-            })
+            })?;
+
+        let was_modified = now == *sled.time_modified();
```

Comment:
Did you check that this exact equality check actually works? I thought Cockroachdb truncates timestamps, so even if we did update this sled, it would be losing precision, and thus not appear equal

(IMO we should prefer using inequality operators with time, if we can frame things that way)

Reply:
I am launching a testbed now to see if the collection fires

Reply:
I got a clippy error, so I'll also rework this then test. I knew we talked about this, but wanted to see what would happen. I thought there was a special DB type for these comparisons we used. I'll look into it.

Reply:
I reworked this to compare seconds since the epoch instead. About to test.